cdm/misc/src/main/java/ucar/nc2/geotiff/GeotiffWriter.java

```diff
@@ -213,17 +300,54 @@ private void writeMetadata(boolean greyScale, double xStart, double yStart, doub
       geotiff.addTag(new IFDEntry(Tag.XResolution, FieldType.RATIONAL).setValue(1, 1));
       geotiff.addTag(new IFDEntry(Tag.YResolution, FieldType.RATIONAL).setValue(1, 1));
       geotiff.addTag(new IFDEntry(Tag.ResolutionUnit, FieldType.SHORT).setValue(1));
-
+      // black is zero (value used in GeotiffWriter)
+      geotiff.addTag(new IFDEntry(Tag.PhotometricInterpretation, FieldType.SHORT).setValue(1));
     } else {
+      if (colorTable != null && colorTable.length > 0) {
+        geotiff.addTag(new IFDEntry(Tag.PhotometricInterpretation, FieldType.SHORT).setValue(3));
+        geotiff.addTag(new IFDEntry(Tag.ColorMap, FieldType.SHORT, colorTable.length).setValue(colorTable));
+      } else {
+        geotiff.addTag(new IFDEntry(Tag.PhotometricInterpretation, FieldType.SHORT).setValue(1)); // black is zero :
+                                                                                                  // not used?
+      }
       // standard tags for SampleFormat ( see TIFF spec, section 19)
-      geotiff.addTag(new IFDEntry(Tag.BitsPerSample, FieldType.SHORT).setValue(32)); // 32 bits per sample
-      geotiff.addTag(new IFDEntry(Tag.SampleFormat, FieldType.SHORT).setValue(3)); // Sample Format
+      geotiff.addTag(new IFDEntry(Tag.BitsPerSample, FieldType.SHORT).setValue(elemSize * 8));
+
+      if (dtype.isIntegral()) {
+        geotiff.addTag(new IFDEntry(Tag.SampleFormat, FieldType.SHORT).setValue(dtype.isUnsigned() ? 1 : 2)); // UINT or
+                                                                                                              // INT
+      } else if (dtype.isFloatingPoint()) {
+        geotiff.addTag(new IFDEntry(Tag.SampleFormat, FieldType.SHORT).setValue(3)); // IEEE Floating Point Type
+      } else {
+        throw new IllegalArgumentException("Unsupported data type for geotiff: " + dtype);
+      }
       geotiff.addTag(new IFDEntry(Tag.SamplesPerPixel, FieldType.SHORT).setValue(1));
-      float min = (float) (dataMinMax.min);
-      float max = (float) (dataMinMax.max);
-      geotiff.addTag(new IFDEntry(Tag.SMinSampleValue, FieldType.FLOAT).setValue(min));
-      geotiff.addTag(new IFDEntry(Tag.SMaxSampleValue, FieldType.FLOAT).setValue(max));
-      geotiff.addTag(new IFDEntry(Tag.GDALNoData, FieldType.FLOAT).setValue(min - 1.f));
```

Comment:
Just realized that this might be technically wrong. According to https://www.awaresystems.be/imaging/tiff/tifftags/gdal_nodata.html, this field is supposed to be ASCII encoded?

Reply:
Should I fix this as part of this PR or submit a separate PR? I'm just now realizing that this mistake has been causing all sorts of low-level "weird" issues that I could never quite figure out.

Reply:
Just came to the realization that there are no pre-existing tests for the geotiff writer with greyscale set to false. I think I'm going to take care of that first along with this nodata fix in a separate PR.

Reply:
Opened https://github.com/Unidata/netcdf-java/pull/1335 to address the NoData issue and to expand the tests a bit. I'll rebase this PR on top of that.